packages/block-editor/src/store/reducer.js

```diff
@@ -678,6 +715,10 @@ const withReplaceInnerBlocks = ( reducer ) => ( state, action ) => {
 			),
 		};
 	}
+	stateAfterInsert.getBlocksCache = {
+		...stateAfterInsert.getBlocksCache,
+		...restoreCacheKeys,
+	};
 	return stateAfterInsert;
 };
```

Comment:
It's very hard to understand chat all of this is trying to do, can we maybe add unit tests to clarify?

Reply:
absolutely!